### DISCUSSION TITLE
[Question]: how to use reverse proxy?

### DISCUSSION THREAD

**Discussion by xczsonn** [danny-avila/LibreChat#3854] [Q&A]
### What is your question?

 First, I apologize for the stupid question. I want to use my reverse proxy with Libre Chat. Risu AI has an option called 'URL Auto-fill' and a separate field for entering the password, making it easy to use, but I'm not sure how to apply it to this program.

For example, if the address of my proxy site is 'https://soupkitchen-sk.hf.space/fdgjmk/openai' and the password is 'I AM STRONG#42', I don't know where to enter this information. I tried applying it in the env file as follows, but it didn't work:

If possible, since I'm just a dumb user, could you please provide me with a format for the env file that I can directly apply by copying and pasting, like the one above?

### More Details

OPENAI_API_KEY=I AM STRONG#42
# OPENAI_MODELS=gpt-4o,chatgpt-4o-latest,gpt-4o-mini,gpt-3.5-turbo-0125,gpt-3.5-turbo-0301,gpt-3.5-turbo,gpt-4,gpt-4-0613,gpt-4-vision-preview,gpt-3.5-turbo-0613,gpt-3.5-turbo-16k-0613,gpt-4-0125-preview,gpt-4-turbo-preview,gpt-4-1106-preview,gpt-3.5-turbo-1106,gpt-3.5-turbo-instruct,gpt-3.5-turbo-instruct-0914,gpt-3.5-turbo-16k

DEBUG_OPENAI=false

# TITLE_CONVO=false
# OPENAI_TITLE_MODEL=chatgpt-4o-latest

# OPENAI_SUMMARIZE=true
# OPENAI_SUMMARY_MODEL=chatgpt-4o-latest

# OPENAI_FORCE_PROMPT=true

# OPENAI_REVERSE_PROXY=https://soupkitchen-sk.hf.space/fdgjmk/openai

# OPENAI_ORGANIZATION=

### What is the main subject of your question?

Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by fuegovic** [comment]
you can configure custom endpoints in the `librechat.yaml` file, see:
[librechat_yaml setup](https://www.librechat.ai/docs/configuration/librechat_yaml/setup)

you first need to modify the docker override file to enable the yaml config, then add your custom proxy in the yaml config
following the instructions in the docs should get you there, you can also look here for more librechat.yaml config examples:
https://github.com/LibreChat-AI/librechat-config-yaml

**Comment by xczsonn** [comment]
> 파일에서 사용자 지정 끝점을 구성할 수 있습니다(참조: [librechat_yaml 설정](https://www.librechat.ai/docs/configuration/librechat_yaml/setup) 참조).`librechat.yaml`
> 
> 먼저 YAML 구성을 활성화하도록 Docker 재정의 파일을 수정한 다음 문서의 지침에 따라 YAML 구성에 사용자 지정 프록시를 추가해야 하며 여기에서 더 많은 librechat.yaml 구성 예제를 찾을 수도 있습니다. https://github.com/LibreChat-AI/librechat-config-yaml

 Is it not enough to fill in the existing OpenAI field? Is it correct to enter my key in the API key field? Or is there a separate variable prepared?

**Comment by fuegovic** [comment]
First of all, you'll probably need a new password (API KEY) without `#` or spaces

If you want to replace the OpenAI endpoint with your own, you can (uncomment if needed and) fill in these .env var at minimum
```yaml
OPENAI_API_KEY=
OPENAI_REVERSE_PROXY=
OPENAI_MODELS=
```
for `OPENAI_REVERSE_PROXY=` use the full url with v1, not the just base URL

The config you shared looks mostly ok, make sure you have a password without spaces or `#` and make sure the URL is complete and it should work
